libs/http-datasource/src/datasource-client.cpp

```diff
@@ -30,8 +35,11 @@ void RemoteDataSource::fill(const TileFeatureLayer::Ptr& featureTile)
 TileFeatureLayer::Ptr
 RemoteDataSource::get(const MapTileKey& k, Cache::Ptr& cache, const DataSourceInfo& info)
 {
+    // Round-robin usage of http clients to facilitate parallel requests
+    auto& client = httpClients_[(nextClient_++) % httpClients_.size()];
+
     // Send a GET tile request
-    auto tileResponse = httpClient_.Get(stx::format(
+    auto tileResponse = client.Get(stx::format(
```

Comment:
How does this work if the client is already in use by another invocation of `::get`, on another thread?

Reply:
The httplib client is internally thread-safe, so it will just wait - this means that the data source server will really never get more parallel requests than its `maxParallelRequests` value indicates.

Reply:
Then this is still suboptimal because if the selected client is busy, but the next one would be free, this will still block. But I think that is ok(?).

Reply:
Yes, I think so. I like the simple round-robin :)